grakn-engine/src/main/java/ai/grakn/engine/controller/api/RelationshipController.java

```diff
@@ -0,0 +1,128 @@
+import static ai.grakn.util.REST.Request.KEYSPACE;
+import static ai.grakn.util.REST.WebPath.Api.RELATIONSHIP;
+import static ai.grakn.util.REST.WebPath.Api.RELATIONSHIP_ENTITY_ROLE_ASSIGNMENT;
+
+/**
+ * <p>
+ *     Relationship endpoints
+ * </p>
+ *
+ * @author Ganeshwara Herawan Hananda
+ */
+
+public class RelationshipController {
+    private final EngineGraknTxFactory factory;
+    private static final Logger LOG = LoggerFactory.getLogger(RelationshipController.class);
+
+    public RelationshipController(EngineGraknTxFactory factory, Service spark) {
+        this.factory = factory;
+
+        spark.post(RELATIONSHIP, this::postRelationship);
+        spark.put(RELATIONSHIP_ENTITY_ROLE_ASSIGNMENT, this::assignEntityAndRoleToRelationship);
+        // TODO: implement it after operation has been supported in the Graph API
+//        spark.delete("/api/relationship/:relationshipConceptId/role/:roleConceptId/entity/:entityConceptId", this::deleteEntityAndRoleToRelationshipAssignment);
+    }
+
+    private Json postRelationship(Request request, Response response) {
+        LOG.info("postRelationship - request received.");
+        String relationshipTypeLabel = mandatoryPathParameter(request, "relationshipTypeLabel");
+        String keyspace = mandatoryQueryParameter(request, KEYSPACE);
+        LOG.info("postRelationship - attempting to find entityType " + relationshipTypeLabel + " in keyspace " + keyspace);
+        try (GraknTx tx = factory.tx(Keyspace.of(keyspace), GraknTxType.WRITE)) {
+            Optional<RelationshipType> relationshipTypeOptional = Optional.ofNullable(tx.getRelationshipType(relationshipTypeLabel));
+            if (relationshipTypeOptional.isPresent()) {
+                LOG.info("postRelationship - relationshipType " + relationshipTypeLabel + " found.");
+                RelationshipType relationshipType = relationshipTypeOptional.get();
+                Relationship relationship = relationshipType.addRelationship();
+                String jsonConceptId = relationship.getId().getValue();
+                LOG.info("postRelationship - relationship " + jsonConceptId + " of relationshipType " + relationshipTypeLabel + " added. request processed");
+                response.status(HttpStatus.SC_OK);
+                return relationshipJson(jsonConceptId);
+            } else {
+                LOG.info("postRelationship - relationshipType " + relationshipTypeLabel + " NOT found.");
+                response.status(HttpStatus.SC_BAD_REQUEST);
+                return Json.nil();
+            }
+        }
+    }
+
+    private Json assignEntityAndRoleToRelationship(Request request, Response response) {
+        LOG.info("assignEntityAndRoleToRelationship - request received.");
+        String relationshipConceptId = mandatoryPathParameter(request, "relationshipConceptId");
+        String roleConceptId = mandatoryPathParameter(request, "roleConceptId");
```

Comment:
More hard coded strings 😢

Reply:
fixed in bd46f4b2c7344d9d48caee578aa1c7d05d2810ed